src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb

```diff
@@ -510,9 +457,65 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
+            Dim stApplicationIcon As String = Nothing
+            Dim stApplicationManifest As String = Nothing
+            If Not PropertyControlData.IsSpecialValue(obj) Then
+
+                stApplicationIcon = TryCast(obj, String)
+
+                If String.IsNullOrEmpty(Trim(stApplicationIcon)) Then
+                    If (ProjectProperties.OutputType <> VSLangProj.prjOutputType.prjOutputTypeLibrary) Then
+                        stApplicationIcon = My.Resources.Microsoft_VisualStudio_Editors_Designer.PPG_Application_DefaultIconText
+                    Else
+                        '// ApplicationIcon can be empty for dlls
+                    End If
+                End If
+            End If
+
+            obj = TryGetNonCommonPropertyValue(GetPropertyDescriptor("ApplicationManifest"))
+            If Not PropertyControlData.IsSpecialValue(obj) Then
+
+                stApplicationManifest = TryCast(obj, String)
+                stApplicationManifest = Trim(stApplicationManifest)
+
+                If String.Equals(stApplicationManifest, prjApplicationManifestValues.prjApplicationManifest_Default, StringComparison.OrdinalIgnoreCase) Then
+                    stApplicationManifest = My.Resources.Microsoft_VisualStudio_Editors_Designer.PPG_Application_DefaultManifestText
+                ElseIf String.Equals(stApplicationManifest, prjApplicationManifestValues.prjApplicationManifest_NoManifest, StringComparison.OrdinalIgnoreCase) Then
+                    stApplicationManifest = My.Resources.Microsoft_VisualStudio_Editors_Designer.PPG_Application_NoManifestText
+                ElseIf String.IsNullOrEmpty(stApplicationManifest) Then
+                    If (ProjectProperties.OutputType <> VSLangProj.prjOutputType.prjOutputTypeLibrary) Then
+                        stApplicationManifest = My.Resources.Microsoft_VisualStudio_Editors_Designer.PPG_Application_DefaultManifestText
+                    Else
+                        '// ApplicationManifest can be empty for dlls
+                    End If
+                End If
+            End If
+
+            ApplicationIcon.Text = stApplicationIcon
+            ManifestExplanationLabel.Enabled = True
+            IconRadioButton.Checked = True
+            Win32ResourceFile.Enabled = False
+            Win32ResourceFileBrowse.Enabled = False
+            Win32ResourceRadioButton.Checked = False
+            EnableControl(ApplicationIconLabel, ApplicationIconSupported())
+            EnableControl(ApplicationIcon, ApplicationIconSupported())
+            EnableControl(AppIconBrowse, ApplicationIconSupported())
+            If (ProjectProperties.OutputType <> VSLangProj.prjOutputType.prjOutputTypeLibrary) Then
+                ApplicationManifest.Text = stApplicationManifest
+                EnableControl(ApplicationManifestLabel, ApplicationManifestSupported())
+                EnableControl(ApplicationManifest, ApplicationManifestSupported())
```

Comment:
Have you tested changing the output type? Are the enabled states of these controls updated correctly?